dspace-api/src/main/java/org/dspace/util/SolrUpgradeStatistics6.java

```diff
@@ -0,0 +1,610 @@
+                        server.commit();
+                        docs.clear();
+                }
+                if (context != null) {
+                        try {
+                                totalCache += numUncache + context.getCacheSize();
+                        } catch (SQLException e) {
+                        }
+                }
+                this.context = new Context(Context.Mode.READ_ONLY);
+                lastItem = null;
+                lastBitstream = null;
+                numUncache = 0;
+        }
+        
+        /*
+         * Compute the number of items that were cached by hibernate since the context was cleared.
+         */
+        private long getCacheCounts(boolean fromStart) {
+                long count = 0;
+                try {
+                        count = context.getCacheSize();
+                } catch (Exception e) {
+                        //no action
+                }
+                count += this.numUncache;
+                if (fromStart) {
+                        count += totalCache;
+                }
+                return count;
+        }
+        
+        /**
+         * Compute the time since the last batch was processed
+         * @param fromStart if true, report on processing time since the start of the program
+         * @return the time in ms since the start time
+         */
+        public long logTime(boolean fromStart) {
+                long ret = 0;
+                long cur = new Date().getTime();
+                if (lastTime == -1) {
+                        startTime = cur;
+                } else if (fromStart){
+                        ret = cur - startTime;
+                } else {
+                        ret = cur - lastTime;
+                }
+                lastTime = cur;
+                return ret;
+        }
+
+        /**
+         * Print a status message appended with the processing time for the operation
+         * @param header Message to display
+         * @param fromStart if true, report on processing time since the start of the program
+         */
+        public void printTime(String header, boolean fromStart) {
+                long dur = logTime(fromStart);
+                System.out.println(String.format("%s (%,6d sec; DB cache: %,9d)", header, dur / 1000, getCacheCounts(fromStart)));
+        }
+
+        /*
+         * Create command line option processor
+         */
+        private static Options makeOptions() {
+                Options options = new Options();
+                options.addOption(HELP_OPTION, "help", false, "Get help on options for this command.");
+                options.addOption(INDEX_NAME_OPTION, "index-name", true,
+                                                 "The names of the indexes to process. At least one is required (default=statistics)");
+                options.addOption(NUMREC_OPTION, "num-rec", true, "Total number of records to update (defaut=100,000).");
```

Comment:
I wanted to give the user some mechanism to throttle the work that this command will perform.  I estimate that it will take 24 hours of processing time to run against our stats records when we upgrade.  num-rec can be set very high if desired.